Users/coafile.rst

```diff
@@ -38,8 +38,95 @@ wide coafile which has the lowest priority. The ``default_coafile`` must
 lie in the coala installation directory and is valid for everyone using
 this coala installation.
 
-Setting Inheritance
--------------------
+Explicit Setting Inheritance
+----------------------------
+
+Every coafile contains one or more sections. Section names are case
+insensitive. The old implicit section inheritance syntax has been deprecated
+and has been scheduled for removeal. Instead, define section inheritance
+explicitly by naming a section in the format ``[basesection.newsection]``.
+Extra values can be appended to an inherited setting using the ``+=``
+operator.
+
+Consider the following coafile::
+
+  [all]
+  enabled = True
+  overridable = 2
+  ignore = ./vendor1
+
+  [all.section1]
+  overridable = 3
+  ignore += ./vendor2
+  other = some_value
+
+  [all.section2]
+  overridable = 4
+  ignore += ./vendor3
+  other = some_value
+
+This is the same file without section inheritance::
+
+  [all]
+  enabled = True
+  overridable = 2
+  ignore = ./vendor1
+
+  [section1]
+  enabled = True
+  overridable = 3
+  ignore = ./vendor1, ./vendor2
+  other = some_value
+
+  [section2]
+  enabled = True
+  overridable = 4
+  ignore = ./vender1, ./vendor3
+  other = some_other_value
+
+All settings must be part of a section, so don't do this for implicit
+inheritance (this is also deprecated behavior)::
+
+  # bad!
+  setting1 = 1
+
+  [section1]
+  # setting1 is inherited
+  setting2 = 2
+
+Instead, make the inheritance explicit::
+
+  # better!
+  [all]
+  setting1 = 1
+
+  [all.section1]
+  # setting1 is inherited
+  setting2 = 2
+
+Please only define inheritance from one section per child section::
+
+  # BAD! Will only inherit from "all", not "child" leaving "child.sub" as the
+  # new section's name
+  [all.child.sub]
+  setting1 += a, b ,c
+
+  # ok, "child" inherits from section "all" and "sub" inherits from "child"
+  # which inherits from "all"
+  [all.child]
+  setting1 += b
+
+  [child.sub]
+  setting1 = a, b, c
+
+Setting Inheritance (deprecated)
```

Comment:
we'd have to version our docs but that's not the right day to do it now, maybe directly after the release.